Source/options.h

```diff
@@ -2,19 +2,22 @@
 
 DEVILUTION_BEGIN_NAMESPACE
 
-typedef struct DiabloOptions {
+class DiabloOptions final {

```

Comment:
We should use `struct` (without typedef) for types like this.

`struct` in C++ is identical to `class` except the default visibility is `public`.

Reply:
> for types such as these.

Can you elaborate a bit on this? What caracteristic exactly are you looking for to decide?

Reply:
All fields are public

Reply:
Or even: has any public fields

Basically it's when something is more of a plain data container than encapsulated data and/or behaviour

Reply:
> All fields are public

So even if there is inheritance at play, it should be a `struct` if it only contains `public` fields?

Reply:
Yes

Reply:
Of course, as soon as you have anything `virtual`, it's no longer a plain data container and so should be a `class`.

C++ programmers expect that `struct` are generally POD types, so e.g. can be copied with a simple memcpy. Inheritance is at odds with that and should be avoided for structs.